### PR TITLE
Improved delegation doc (adding .well-known info)

### PR DESCRIPTION
**Pull request by vaab**
Just a suggestion of improvement of the doc provided in the README... that's by no mean a perfect solution and am happy if it is not integrated for some reason, but at least if this kind of doc could have been there in the first place, I would definitively saved one days of battling my way through the numerous incomplete documentations and guide.

It this addition seems to help, I'll add the changelog file gladly for inclusion.

### Pull Request Checklist

* [X] Pull request is based on the develop branch
* [need request number] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [X] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

**Comment by vaab** [comment]
It definitively needs technical overview, as I'm very new to ``matrix`` and ``synapse``, and basic wording as I'm not a native english speaker.

**Comment by neilisfragile** [comment]
Hi @vaab this is great, thanks so much for putting the time in :+1:

I'm actually trying to solve a similar problem problem at https://github.com/matrix-org/synapse/pull/4766 I'll either incorporate your work into 4766 (and ensure that you are credited) or make some changes on this PR before merging.

**Comment by neilisfragile** [comment]
Tracking here https://github.com/matrix-org/synapse/pull/4832

File: README.rst [hunk 1]
```diff
@@ -254,41 +339,36 @@ its user-ids, by setting ``server_name``::
...
-  (it should be ``_matrix._tcp.<server_name>``), and that the port and hostname
-  it specifies are reachable from outside your network.
-
 Another common problem is that people on other servers can't join rooms that
 you invite them to. This can be caused by an incorrectly-configured reverse
 proxy: see `<docs/reverse_proxy.rst>`_ for instructions on how to correctly
 configure a reverse proxy.
 
+
 Running a Demo Federation of Synapses
```
Comment (richvdh) [inline README.rst]: let's get rid of this if it's moved to federate.md?

Comment (richvdh) [reply README.rst]: oops I'm looking at the wrong PR, sorry

**Comment by neilisfragile** [comment]
@vaab #4832 now merged - I had to rebase and squash merge, but have provided credit through the commit and changelog entry. Thanks for all your help - the docs are much better now.

**Comment by vaab** [comment]
Thanks ! I noticed you nearly completely removed my (perhaps too heavy) warnings about DNS SRV method not actually delivering a complete delegation as .well-known method would do. This is a keypoint that is quite unexpected and completely invalidates DNS SRV as a working delegation solution in a common scenario : people using virtualhosts, and using both domain name (the delegatee and the delegated) in 2 different virtualhosts that have to answer 2 different SSL ids. In the  pushed documentation, you still sell both delegation method as equivalent, which they are clearly not functionally speaking.

It appeared to me primordial to add this info in the introduction so as the people could safely choose the solution they wanted to use.

To be honest, the time I lost on that particular point was what pushed me trying to contribute here on a better doc, as I really felt misguided.

I'm just trying here to share what seemed important for me in my contribution, I'm very happy that the documentation moved forward anyway. I noticed your careful attention for reviewing my flawed contribution and make it better. And I wish I had more time to bring up this point at the right moment.

Feel free to ignore my point if you feel so.

**Comment by richvdh** [comment]
@vaab: the point is that the two delegation methods offer different features; it's simply not true to say that .well-known is a "complete" delegation and that SRV is somehow incomplete. Arguably .well-known is incomplete because it doesn't offer round-robin and failover support.

Neil's rewrite fairly clearly spells out the requirements for the TLS certificate: [here](https://github.com/matrix-org/synapse/blob/master/docs/federate.md#dns-srv-delegation) "This method requires the target server to provide a valid TLS certificate for the original `server_name`."

**Pull request closed without merging**